app/models/spree/gateway/amazon.rb

```diff
@@ -145,6 +145,16 @@ def void(response_code, gateway_options)
       return ActiveMerchant::Billing::Response.new(true, "Success", response)
     end
 
+    def cancel(response_code)
+      payment = Spree::Payment.find_by(response_code: response_code)
+      order = payment.order
+      load_amazon_mws(payment.source.order_reference)
+
+      response = @mws.refund(response_code, order.number, order.total, order.currency)
```

Comment:
FYI I don't think `order.total` will always be the correct thing to use here.  But since that's what `#void` is already using I think that would be fine to leave for another PR.  I've opened an issue for that here: https://github.com/boomerdigital/solidus_amazon_payments/issues/34